packages/@aws-cdk/aws-appmesh/lib/virtual-gateway.ts

```diff
@@ -0,0 +1,528 @@
+import { AccessLog, HealthCheck, PortMapping, Protocol } from './shared-interfaces';
+
+/**
+ * Interface which all Virtual Gateway based classes must implement
+ */
+export interface IVirtualGateway extends cdk.IResource {
+  /**
+   * Name of the VirtualGateway
+   *
+   * @attribute
+   */
+  readonly virtualGatewayName: string;
+
+  /**
+   * The Amazon Resource Name (ARN) for the VirtualGateway
+   *
+   * @attribute
+   */
+  readonly virtualGatewayArn: string;
+
+  /**
+   * The mesh which the VirtualGateway belongs to
+   */
+  readonly mesh: IMesh;
+
+  /**
+   * Utility method to add a list of listeners to this VirtualGateway
+   */
+  addListeners(listeners: VirtualGatewayListener[]): void;
+
+  /**
+   * Utility method to add a single listener to this VirtualGateway
+   */
+  addListener(listener: VirtualGatewayListener): void;
+
+  /**
+   * Utility method to add a new GatewayRoute to the VirtualGateway
+   */
+  addGatewayRoute(id: string, route: GatewayRouteBaseProps): GatewayRoute;
+}
+
+/**
+ * Represents the properties needed to define HTTP Listeners for a VirtualGateway
+ */
+export interface HttpGatewayListenerProps {
+  /**
+   * Port to listen for connections on
+   *
+   * @default - 8080
+   */
+  readonly port?: number
+
+  /**
+   * The health check information for the listener
+   *
+   * @default - no healthcheck
+   */
+  readonly healthCheck?: HealthCheck;
+}
+
+/**
+ * Represents the properties needed to define GRPC Listeners for a VirtualGateway
+ */
+export interface GrpcGatewayListenerProps {
+  /**
+   * Port to listen for connections on
+   *
+   * @default - 8080
+   */
+  readonly port?: number
+
+  /**
+   * The health check information for the listener
+   *
+   * @default - no healthcheck
+   */
+  readonly healthCheck?: HealthCheck;
+}
+
+/**
+ * Represents the properties needed to define listeners for a VirtualGateway
+ */
+export abstract class VirtualGatewayListener {
+  /**
+   * Returns an HTTP Listener for a VirtualGateway
+   */
+  public static httpGatewayListener(props?: HttpGatewayListenerProps): VirtualGatewayListener {
+    return new HttpGatewayListener(props);
+  }
+
+  /**
+   * Returns an HTTP2 Listener for a VirtualGateway
+   */
+  public static http2GatewayListener(props?: HttpGatewayListenerProps): VirtualGatewayListener {
+    return new Http2GatewayListener(props);
+  }
+
+  /**
+   * Returns a GRPC Listener for a VirtualGateway
+   */
+  public static grpcGatewayListener(props?: GrpcGatewayListenerProps) {
+    return new GrpcGatewayListener(props);
+  }
+
+  /**
+   * Called when the GatewayListener type is initialized. Can be used to enforce
+   * mutual exclusivity
+   */
+  public abstract bind(scope: cdk.Construct): CfnVirtualGateway.VirtualGatewayListenerProperty;
+}
+
+/**
+ * Represents the properties needed to define an HTTP Listener for a VirtualGateway
+ */
+export class HttpGatewayListener extends VirtualGatewayListener {
```

Comment:
This doesn't need to be exported I think...?

Reply:
Good call. Removed the export